### PR TITLE
fix: Load issues with changes to lazy mode/extension loading

### PR DESCRIPTION
**Pull request by wayfarer3130**
### Context

There are three changes being made in this PR to resolve the issues with the lazy loading:

* Fix the Mode.tsx loading timeline to take into account the end of loading the modules in more places
* Fix the loading of modules so that the service itself can handle the module init in order to load default values after service init
* Add documentation and configuration for a custom PLUGIN_CONFIG file to allow updating the plugin config (this was supposed to have been added to the original change but got missed - it allows custom versions of the plugins without necessary changing any code in OHIF)

### Changes & Results

Before this PR, the render methods would start too early on the Mode init, causing renders to fail.  As well, the service init on the customization service was handling the load of default modules, which was too early to complete the load, so this caused Mode loaded modules to not get initialized.

### Testing

Test loading the segmentation, exiting the mode and coming back in and loading the SEG/RT again.

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

**Comment by sedghi** [comment]
I will close this PR in favor of the other PR that works for now until you get back to this and fix the rest of the issues https://github.com/OHIF/Viewers/pull/3377/files

File: platform/core/src/services/HangingProtocolService/HangingProtocolService.ts [hunk 1]
```diff
@@ -1524,4 +1525,14 @@ export default class HangingProtocolService extends PubSubService {
   _copyProtocol(protocol: Protocol) {
     return JSON.parse(JSON.stringify(protocol));
   }
+
+  /** Register the hanging protocols from the extension module */
+  public initModule(extensionModule, extensionId: string): void {
```
Comment (wayfarer3130) [inline platform/core/src/services/HangingProtocolService/HangingProtocolService.ts:1530]: extensonModule => hangingProtocolModule
remove extensionId - it is still there, but unused

File: platform/core/src/services/CustomizationService/CustomizationService.ts [hunk 2]
```diff
@@ -87,6 +88,16 @@ export default class CustomizationService extends PubSubService {
     });
   }
 
+  /** Register the customizations from the extensions */
+  public initModule(extensionModule, extensionId: string): void {
```
Comment (wayfarer3130) [inline platform/core/src/services/CustomizationService/CustomizationService.ts:92]: Remove extensionId

File: platform/core/src/extensions/ExtensionManager.ts [hunk 3]
```diff
@@ -312,6 +318,20 @@ export default class ExtensionManager {
...
+      if (!element.name) {
+        throw new Error(
+          `Extension ID ${extensionId} module ${moduleType} element has no name`
+        );
+      }
+      const id = `${extensionId}.${moduleType}.${element.name}`;
+      element.id = id;
+      this.modulesMap[id] = element;
+    });
+    service?.initModule?.(extensionModule, extensionId);
```
Comment (wayfarer3130) [inline platform/core/src/extensions/ExtensionManager.ts:332]: service?.onServiceModuleRegistration?.(....

**Comment by sedghi** [comment]
we should continue on this later

**Comment by wayfarer3130** [comment]
Will reconsider this in the future with a more extensive change

**Pull request closed without merging**